stake-pool/js/src/utils/stake.ts

```diff
@@ -186,35 +185,25 @@ export async function prepareWithdrawAccounts(
 /**
  * Calculate the pool tokens that should be minted for a deposit of `stakeLamports`
  */
-export function calcPoolTokensForDeposit(stakePool: StakePool, stakeLamports: number): number {
+export function calcPoolTokensForDeposit(stakePool: StakePool, stakeLamports: BN): BN {
   if (stakePool.poolTokenSupply.isZero() || stakePool.totalLamports.isZero()) {
     return stakeLamports;
   }
-  return Math.floor(
-    divideBnToNumber(new BN(stakeLamports).mul(stakePool.poolTokenSupply), stakePool.totalLamports),
-  );
+  const numerator = stakeLamports.mul(stakePool.poolTokenSupply);
+  return numerator.div(stakePool.totalLamports);
 }
 
 /**
  * Calculate lamports amount on withdrawal
  */
-export function calcLamportsWithdrawAmount(stakePool: StakePool, poolTokens: number): number {
-  const numerator = new BN(poolTokens).mul(stakePool.totalLamports);
+export function calcLamportsWithdrawAmount(stakePool: StakePool, poolTokens: BN): BN {
+  const numerator = poolTokens.mul(stakePool.totalLamports);
   const denominator = stakePool.poolTokenSupply;
   if (numerator.lt(denominator)) {
-    return 0;
-  }
-  return divideBnToNumber(numerator, denominator);
-}
-
-export function divideBnToNumber(numerator: BN, denominator: BN): number {
-  if (denominator.isZero()) {
-    return 0;
+    return new BN(0);
   }
-  const quotient = numerator.div(denominator).toNumber();
-  const rem = numerator.umod(denominator);
-  const gcd = rem.gcd(denominator);
-  return quotient + rem.div(gcd).toNumber() / denominator.div(gcd).toNumber();
+  // ceiling the calculation by adding denominator - 1 to the numerator
+  return numerator.add(denominator).sub(new BN(1)).div(denominator);
```

Comment:
a test to make sure the ceil math works would be good

Reply:
Looking back through, I've realized that this is inconsistent with the program, and will revert: https://github.com/solana-labs/solana-program-library/blob/c38a1b5d6a1a6049b5844aef8e50e889e18918ed/stake-pool/program/src/state.rs#L178-L189

Reply:
Weirdly enough, the result is unused where we `Math.ceil`, so this was never important